homeassistant/components/configurator.py

```diff
@@ -105,14 +133,16 @@ def __init__(self, hass):
         self._cur_id = 0
         self._requests = {}
         hass.services.async_register(
-            DOMAIN, SERVICE_CONFIGURE, self.handle_service_call)
+            DOMAIN, SERVICE_CONFIGURE, self.async_handle_service_call)
 
-    def request_config(
+    @async_callback
+    def async_request_config(
```

Comment:
I wonder how much of this we can't just move into the top level methods and just use `hass.data` for storage and get rid of the instance.